### PR TITLE
Milestone: Todo list project

### PR DESCRIPTION
**Pull request by newhorizon-tech**
This is a mini-project that allows you to :

- Create new tasks for a to-do list
- Display the current tasks in the todo list
- Display different projects(categories)
- Modify an existing task
- Delete a task from the list 

The project utilizes :

- Factory pattern
- DOM manipulation 
- LocalStorage
- ES6 Arrow functions 
- Functions like map and filter 
- ES6 Modules 
- Module and stylesheets bundling

### CODE REVIEW

File: src/display.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    container.append(heading, listElement);
+    projectsMain.appendChild(container);
+
+    const listItems = project.list;
+    listItems.forEach(item => {
+      const listItem = document.createElement('li');
+      const customDate = formatDate(item.duedate);
+      // Saturday, September 17, 2016
+      listItem.textContent = `${item.title}, Date: ${customDate}, Priority: ${item.priority}`;
+      // const testBtn = document.createElement('button');
```
Comment (addod19) [inline src/display.js]: Kindly delete commented code, use git history instead

File: src/display.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      modifyBtn.setAttribute('class', ' btn btn-primary');
+      deleteBtn.innerHTML = '<img src="https://img.icons8.com/material-sharp/15/000000/delete-forever.png" />';
+      deleteBtn.setAttribute('class', 'delete-btn btn btn-danger');
+
+      // const expand = document.querySelector('modifyBtn');
+      // const loose = document.querySelector('deleteBtn');
+
+      // if (modifyBtn != null ){
+      //   modifyBtn.addEventListener('onclick', modifyItem);
+      // }
```
Comment (addod19) [inline src/display.js]: Kindly delete commented code, use git history instead

File: src/display.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      modifyBtn.onclick = () => modifyItem(item, project);
+      deleteBtn.onclick = () => deleteItem(item, project);
+
+      // loose.addEventListener('onclick', deleteItem);
```
Comment (addod19) [inline src/display.js]: Kindly delete commented code, use git history instead

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+// create the input due date
+// const formatDate = (input) => {
+//   const options = {
+//     year: 'numeric', month: 'long', day: 'numeric',
+//   };
+
+//   const date = new Date(input);
+//   const result = date.toLocaleDateString('en-US', options);
+//   return result;
+// };
```
Comment (addod19) [inline src/index.js]: Kindly delete commented code, use git history instead

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+// Adds a task to the project (project is a string)
+const addTaskToProject = (task, project) => {
+  const currentProject = projects.find(o => o.name === project);
+  currentProject.list.push(task);
+  saveData(projects, id);
+  displayProjects(projects);
+};
+
+//  store the list of tasks
+// let listElement = [];
```
Comment (addod19) [inline src/index.js]: Kindly delete commented code, use git history instead

File: src/styles.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  margin-top: 10px;
+}
+
+.project-box ul li {
+  margin: 0.5% 0;
+}
+
+.project-box ul li button {
+  // float: right;
+  // margin-right: 20px;
```
Comment (addod19) [inline src/styles.scss]: Kindly delete commented code, use git history instead

File: package.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "to-do-list",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (addod19) [inline package.json]: Kindly update the script so we can run commands like npm build, dev or start
```
 "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
```

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :trophy: 

Hi @newhorizon-tech and @dasileker, congratulations on this project :muscle: The README file is professional, commit messages easy to understand, PR is detailed with information :slightly_smiling_face: Let us kindly address the comments below so we can move ahead:

- The task should also be an object, currently I cannot see it declared as an object
- Kindly update the scripts as in the comments below
- Kindly remove all commented codes

[Optional]
- Kindly add basic validations so a user cannot submit empty data which could be expensive for our application

Once these are fixed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by newhorizon-tech** [comment]
We made the following changes: 

- Added NPM scripts 
- Improved project styling 
- Added validation

**Review by SpaYco — APPROVED** [review]
# Project Approved

thank you for making the required changes, your project is now approved!

### please don't forget to merge to master, good luck with the next projects!


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Pull request merged**